### ISSUE TITLE
Help with example of working config for "Address by Ada Handle" reducer

### ISSUE THREAD

**Issue by WesleyBatista** [txpipe/scrolls#96]
Could anyone help me testing this config in preprod environment? Or maybe provide an example of a working one?

```toml
[source]
type = "N2N"
address = "preprod-node.world.dev.cardano.org:30000"

[[reducers]]
type = "AddressByAdaHandle"
key_prefix = "AddressByAdaHandle"
policy_id_hex = "f0ff48bbb7bbe9d59a40f1ce90e9e9d0ff5002ec48f232b49ca0fb9a"

[storage]
type = "Redis"
connection_params = "redis://127.0.0.1:6379/2"

[chain]
type = "PreProd"

[intersect]
type = "Point"
value = [8261225,"103ff3cfec6e388db803fc10dbdecb3026b4a51382008d01ff3f9121be6fa6e4"]
```

After minting this handle [`$myhandle`](https://testnet.cardanoscan.io/token/f0ff48bbb7bbe9d59a40f1ce90e9e9d0ff5002ec48f232b49ca0fb9a6d7968616e646c65):
![image](https://user-images.githubusercontent.com/755254/191995171-7b23ebd1-5c32-4dcd-9487-792f38983907.png)

I was expecting to see it under the key `AddressByAdaHandle.$myhandle`, in my redis logical db 2, but nothing shows up:
```
[2] > keys *
1) "_cursor"
```

I am using the binary from [asset_holders_by_asset_id](https://github.com/txpipe/scrolls/compare/asset_holders_by_asset_id...main) branch which already has the reducer implemented. FYI, I've built it with `cargo build --release --locked --all-features && strip ./target/release/scrolls || true` and I am calling like this: `/usr/local/bin/scrolls daemon --config /path/to/config.toml`. I also do `del _cursor` on redis before each test.

___

_oura command to help debugging:_
```
oura watch --since 8261225,103ff3cfec6e388db803fc10dbdecb3026b4a51382008d01ff3f9121be6fa6e4 preprod-node.world.dev.cardano.org:30000 --bearer tcp --magic preprod`
```

Thanks for looking! :bouquet:

**Comment by scarmuega** [comment]
hi @WesleyBatista, thanks for reporting. Your setup seems correct. I'll try to reproduce on my side.

FYI, I've successfully ran this reducer on mainnet and I it yields several thousands of entries.

Quick question: have you ran any other reducer from your setup? Wondering if this is something specific to adahandles or not.

**Comment by WesleyBatista** [comment]
thanks a lot!

yes I did try other reducers, and the ones I tested all seemed fine, except for the adahandles one:
```toml
# get data from a relay node
[source]
type = "N2N"
address = "preprod-node.world.dev.cardano.org:30000"

[policy]
missing_data = "Skip"

[enrich]
type = "Sled"
db_path = "/opt/scrolls/sled_db_preprod"

[[reducers]]
type = "UtxoByAddress"
key_prefix = "UtxoByAddress"
[[reducers]]
type = "PointByTx"
key_prefix = "PointByTx"
[[reducers]]
type = "AddressByTxo"
key_prefix = "AddressByTxo"
[[reducers]]
type = "BalanceByAddress"
key_prefix = "BalanceByAddress"
[[reducers]]
type = "TxByHash"
key_prefix = "TxByHash"
[[reducers]]
type = "TxCountByAddress"
key_prefix = "TxCountByAddress"
[[reducers]]
type = "AddressByAdaHandle"
key_prefix = "AddressByAdaHandle"
policy_id_hex = "f0ff48bbb7bbe9d59a40f1ce90e9e9d0ff5002ec48f232b49ca0fb9a"
[[reducers]]
type = "LastBlockParameters"
key_prefix = "LastBlockParameters"
[[reducers]]
type = "AssetHoldersByAsset"
key_prefix = "AssetHoldersByAsset"


[storage]
type = "Redis"
connection_params = "redis://127.0.0.1:6379/2"

[chain]
type = "PreProd"

[intersect]
type = "Origin"
```
![scrolls-running](https://user-images.githubusercontent.com/755254/192108469-1a7120b8-37d5-417f-bd9c-efda20f38992.gif)

_The `policy` setting is something I tried based on your troubleshooting for another problem I found on discord, but didn't work here._

**Comment by WesleyBatista** [comment]
@scarmuega as you said the reducer worked on mainnet I tried this config:
```toml
# daemon-test-config.toml
[source]
type = "N2N"
address = "relays-new.cardano-mainnet.iohk.io:3001"

[[reducers]]
type = "AddressByAdaHandle"
policy_id_hex = "f0ff48bbb7bbe9d59a40f1ce90e9e9d0ff5002ec48f232b49ca0fb9a"

[storage]
type = "Redis"
connection_params = "redis://127.0.0.1:6379/2"

[chain]
type = "Mainnet"

[intersect]
# type = "Origin"
type = "Point"
value = [72476575,"45a3286d6438799156172487480452512d83df63670c4f719d38f0eeded68083"]
```

  ...expecting to see [$glaedr](https://cardanoscan.io/transaction/0a93c750097ee30123b9ec85fb35af16d5b67ff58ea36b64a82c1d8b7b8bae76?tab=tokenmint) for instance, or anything after that transaction like `$shruikan` or `$thorn`, but nothing...

fyi, this is the output I get:
![image](https://user-images.githubusercontent.com/755254/192113766-9461931d-a30d-4133-a4ad-084b12d179f3.png)

_command in the image: `RUST_BACKTRACE=1 RUST_LOG=INFO ./target/release/scrolls daemon --config daemon-test-config.toml`_

**Comment by scarmuega** [comment]
@WesleyBatista I found the issue, it was quite dumb: the logic for the reducer wasn't taking into account Babbage Txs 🤦.

We've fixed this on #98, which is already merged into main. I've tried your example using that branch and I get the valid Redis entry.

Closing the issue, but feel free to re-open if there're any other questions.

**Comment by WesleyBatista** [comment]
I can confirm that main branch is working with preprod environment :heavy_check_mark: 

![image](https://user-images.githubusercontent.com/755254/192845884-a082e85c-d9b9-4fb0-9b03-70741cfba2de.png)

Thanks a lot, that was quick! :rocket:
